build-system/pr-check/dist.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script builds the AMP runtime for production.
+ * This is run during the CI stage = build; job = dist.
+ */
+
+const colors = require('ansi-colors');
+const {
+  printChangeSummary,
+  startTimer,
+  stopTimer,
+  timedExecOrDie: timedExecOrDieBase,
+  uploadDistOutput} = require('./utils');
+const {isTravisPullRequestBuild} = require('../travis');
+const FILENAME = 'dist.js';
+const FILELOGPREFIX = colors.bold(colors.yellow(`${FILENAME}:`));
+const timedExecOrDie =
+  (cmd, unusedFileName) => timedExecOrDieBase(cmd, FILENAME);
+
+function main() {
+  const startTime = startTimer(FILENAME, FILENAME);
+
+  if (!isTravisPullRequestBuild()) {
+    timedExecOrDie('gulp dist --fortesting');
+    uploadDistOutput(FILENAME);
+  } else {
+    printChangeSummary();
```

Comment:
This line should only be called when something useful is being done. I suppose this should be moved to the `if` case above?

Reply:
Easy change here, but it will get a lot messier if we want to keep it consistent across the jobs. For example for PR builds in `pr-check/local-tests.js`, there are 4 different if statements with no clean way to know when to start the timer.

Okay to keep it as is?

Reply:
I hadn't thought of the timer. SGTM.